03-Bash/fix_owner/fix_owner.sh

```diff
@@ -0,0 +1,76 @@
+#!/bin/bash 
+
+declare -i force
+force=0
+
+function printHelp()
+{
+    echo "$0 - check user\`s"
+    echo "-f, --force - fix without asknowledgement,"
+    echo "-h, --help - print these options."
+    exit 0
+}
+
+# check options
+if [ $# -gt 0 ]; then
+    case $1 in 
+        -h | --help) printHelp;; 
+        -f | --force) force=1;;
+        *) exit 1;;
+    esac
+fi
+
+
+file="/etc/passwd"
+IFS=$'\n'
+for user_line in $(cat $file)
+do
+    declare -i user_id
+    user_id=$(echo $user_line | cut -d ':' -f3)
+    if [ $user_id -lt 1000 ]; then
+        continue
+    fi
+    user_name=$(echo $user_line | cut -d ':' -f1)
+    user_home=$(echo $user_line | cut -d ':' -f6)
+    user_group=$(echo $user_line | cut -d ':' -f4)
+
+    # echo "$user_name=$user_id:$user_group-$user_home"
+
+    homeFileList=$(mktemp)
+    find $user_home -maxdepth 1 > "$homeFileList"
+
+    declare -i once
+    once=0
+    for homeFile in $(cat $homeFileList)
+    do
+        if [ $once -eq 0 ]; then 
+            once=1
+            continue
+        fi
+
+        fileOwner=$(stat -c '%U' $homeFile)
+        if [ "$fileOwner" != "$user_name" ]; then
```

Comment:
Fixing group id is also required for this task.

Reply:
Routine added